src/services/EmployeeService.ts

```diff
@@ -40,27 +80,154 @@ class EmployeeService {
 
   public update(employee: EmployeeBaseModel, id: string): Promise<EmployeeModel> {
     return this.api.put<Employee>(`/api/employee/${id}`, new EmployeeRequestModel(employee))
-      .then((response: Employee) => new EmployeeModel(response));
+      .then((response: Employee) => new EmployeeModel(response))
+      .catch((error) => {
+        if (error.status === ResponseStatusCode.Unauthorized) {
+          throw new ServiceError('Unauthenticated or invalid token');
+        }
+
+        if (error.status === ResponseStatusCode.Forbidden) {
+          throw new ServiceError('Not allowed to update employees');
+        }
+
+        if (error.status === ResponseStatusCode.NotFound) {
+          throw new ServiceError('Employee not found');
+        }
+
+        if (error.status === ResponseStatusCode.PreconditionFailed) {
+          throw new ServiceError('Precondition for updateing employee failed');
+        }
+
+        if (error.status === ResponseStatusCode.InternalServerError) {
+          throw new ServiceError('Internal server error');
+        }
+
+        if (error.status === ResponseStatusCode.NetworkError) {
+          throw new ServiceError('Error contacting server');
+        }
+
+        throw error;
+      });
   }
 
   public delete(id: string): Promise<void> {
     return this.api.delete<void>(`/api/employee/${id}`)
-      .then(() => {});
+      .then(() => {})
+      .catch((error) => {
+        if (error.status === ResponseStatusCode.Unauthorized) {
+          throw new ServiceError('Unauthenticated or invalid token');
+        }
+
+        if (error.status === ResponseStatusCode.Forbidden) {
+          throw new ServiceError('Not allowed to delete employees');
+        }
+
+        if (error.status === ResponseStatusCode.NotFound) {
+          throw new ServiceError('Employee not found');
+        }
+
+        if (error.status === ResponseStatusCode.InternalServerError) {
+          throw new ServiceError('Internal server error');
+        }
+
+        if (error.status === ResponseStatusCode.NetworkError) {
+          throw new ServiceError('Error contacting server');
+        }
+
+        throw error;
+      });
   }
 
   public createContract(contract: ContractBaseModel): Promise<ContractModel> {
     return this.api.post<Contract>('/api/contract', new ContractRequestModel(contract))
-      .then((response: Contract) => new ContractModel(response));
+      .then((response: Contract) => new ContractModel(response))
+      .catch((error) => {
+        if (error.status === ResponseStatusCode.Unauthorized) {
```

Comment:
Add the Unauthorized, InternalServerError and Network error checks in there.
Put it as a static function into ApiService.ts

```
checkDefaultResponseStatus(error);
```